### DISCUSSION TITLE
ubiquity Feature

### DISCUSSION THREAD

**Discussion by rrreza** [phpMv/ubiquity#167] [General]
ubiquity is very good framework. but not famous because:
1- The name "Ubiquity" is Not Unique and cute name.
2- The document and guide of it is not enough. we need Video in YouTube and more article.

.

**Comment by jcheron** [comment]
Hi @rrreza 
I fully agree with these 2 points.
Ubiquity is a very bad name for the reasons you gave.
I think it is never too late to rename it, especially since it is not yet well known. You are free to make proposals.

Concerning documentation and articles, I am working on it. But time is not extensible.
It is certainly wrong for me to focus on developing new features, rather than documenting existing ones.
In this case too, all contributions are welcome, even if they are minimal:
- some [lines](https://github.com/phpMv/ubiquity/tree/master/docs) of [documentation](https://micro-framework.readthedocs.io/en/latest/)
- some code examples in a [demo](https://github.com/phpMv/ubiquity-demos) application
- Opening an [issue](https://github.com/phpMv/ubiquity/issues/new/choose) to report a problem or ask a question as you just did.

It is also a good thing to contribute from the outside:
You can ask your questions on [stackoverflow](https://stackoverflow.com/search?q=ubiquity+php) and php communities.
A framework without a community remains an unknown framework and can never be a good framework.

**Comment by zilveer** [comment]
Why not change framework name to something like Leopard or Jaguar, since Ubiquity's uniqueness is it's speed I think one of the names Jaguar or Leopard would fit here very good !

I vote for Jaguar!

Regards Zilveer

**Comment by jcheron** [comment]
You're not good in zoology @zilveer .
it's the fastest cheetah -> Cheetah Framework ?
And Jaguar is already taken by the english cars.

Otherwise, the ostrich is fast too.
How about Ostrich Framework?

**Comment by zilveer** [comment]
@jcheron actually Jaguar is easier to remember or else I do know that Cheetah is the fastest hence my suggestion.

But does it matter if it is taken by a car model ?

Ostrich, the name is hard to spell and hard to remember . I will come with some more suggestions later today :)


Regards

**Comment by jcheron** [comment]
It was humor, I know you know that.
Besides, I don't know why the Cheetah has a rather ridiculous name in English, maybe it's because we gave his name to [Tarzan's monkey](https://en.wikipedia.org/wiki/Cheeta). In French, it's more stylish: **Guépard**.

More seriously, I (badly) chose Ubiquity because given its efficiency (ease of use en performances), the framework allows the developer to focus on the business aspect more than on the technique, which gives him the power to be everywhere.
I would have liked to change (to avoid using something not unique), but in the same spirit.

I was thinking of a contraction of Ubiquity: **Uby**
In France, it is a [wine estate](http://www.domaine-uby.com), and Uby makes very good Armagnacs
 ;-)
But that's another thing: https://en.wikipedia.org/wiki/UBY :-(

It is also to avoid losing the small audience that the framework has been able to acquire.

**Comment by zilveer** [comment]
@jcheron yes, and in Swedish it is called Geopard :)

But honestly Cheetah wasn't a bad name either , so perhaps changing the name to Cheetah ?

Uby is short yes but it doesn't sound good :/

Regards

**Comment by jcheron** [comment]
Except that Cheetah, in addition to being a big cat who runs fast, is also a fairly well known Python template engine...

Uby sounds good in French, it is also the cry of the female [marsupilami](http://www.leg20.com/portfolio-view/marsupilami-la-fleur/).
But if you tell me it doesn't sound good internationally, it's better to avoid it.

I think we're going to have to dig deeper.

**Comment by zilveer** [comment]
Oh didn't think about that it was a python template engine.

Yeah we will need more suggestions, we can have Uby in one of them ;)

**Comment by Tiippex** [comment]
I think Uby is to close to Ruby

**Comment by jcheron** [comment]
And in more, an amputated Ruby :(

**Comment by zilveer** [comment]
What about Melinda Framework, I cannot see it being used anywhere in programming languages ? 
It is simple to spell and not that long either ?

**Comment by jcheron** [comment]
It's not very well known, but it's still taken: [Melinda Framework](http://www.google.fr/search?q=Melinda+Framework)
Do you know a Melinda who runs fast, @zilveer ?

**Comment by zilveer** [comment]
Oh i never pressed on Comment button.
Nah, actually it is a pretty cute name :p @jcheron

**Comment by jcheron** [comment]
@zilveer 
Melinda is a little too cute name, and frameworks are sometimes roughed up.

Proposal of the day: [Ubyss Framework](http://www.google.fr/search?q=Ubyss+Framework)

- Mixture of Ulysses and abyss
- not used
- not close to another framework name
- quite close to Ubiquity

I don't know if that sounds good, and what it means in other countries/languages...

**Comment by n0nag0n** [comment]
How about "Chateau"? Rolls off the tongue..... it's unique, but still readable in English (we have street names with "Chateau" in them). Chateau means castle and castles are cool...

**Comment by jcheron** [comment]
In our country, **Château**, it's either that you have a very big house, with towers,
or that you have a winery (Saint-Emilion, Pomerol, etc.)...
I would not have thought of using it in this context.
Thanks for the proposition, but I'm worried it might sound a little presumptuous.

**Comment by n0nag0n** [comment]
Maybe the easiest thing at this point is for you to get a very short bullet point, almost of your mission statement of your framework, and then find something based off of that. From reading over your framework, what I've gathered is you:

- Use practicality over convention
- Performance is paramount
- Business needs over programmer needs
- More.....?

**Comment by jcheron** [comment]
It is already a good summary, even if practicality and conventions are not necessarily contradictory, and that they should rather join and complement each other.

**Comment by UlasSAYGINIM** [comment]
how about Liber8 ? it will help people to liberate the performance.

**Comment by jcheron** [comment]
@rrreza 
Some videos are online.
I'm working on the subject
https://www.youtube.com/channel/UCSn_VzC0QpXCHmGDrw-_REw/

**Comment by Igorut** [comment]
@jcheron  What do you think about **Presto**? In Italian, _Presto_ means _soon, quickly, shortly_

**Comment by jcheron** [comment]
**Presto** is cool, I don't know how it goes for English speakers.
In the same vein, **Illico** is not bad either.

**Comment by zilveer** [comment]
@jcheron Preso is easier as well as simpler to both write and spell.
My vote goes to Presto ;)

Regards

**Comment by jcheron** [comment]
**Ugly-Duckling Framework** would be a suitable name. The framework is young, and small, both in what it offers (the range of associated features), and in its community, which is microscopic. It shakes up the established rules (some PSR, the all-object, the systematic use of certain concepts or design patterns...). And it's a winning bet, since the performances are there.

However, the php community, influenced by a certain nomenclatura, has difficulties to adhere.
The only echoes of Ubiquity on social networks have been published on Reddit by people who obviously never tried to test Ubiquity, and stopped at a few uninteresting appearances (see [reddit haters](https://www.reddit.com/r/PHP/comments/bok2v9/ubiquity_framework_ideas_experiences/)).

Ubiquity assumes this role of ugly duckling, provided that the web development in php progresses. But the extension of the community and the rallying of developers are a necessity.

Coming back to **Presto**, on reflection, I think it's not necessarily a good idea to promise performance in the name, but just to allow it to be achieved in production.

![image](https://user-images.githubusercontent.com/2511052/83354300-113ecf80-a358-11ea-8d99-5f8130024890.png)

[edit]
If you are interested in simplicity, **Ugly-Duckling Framework** can be the full name, with possible shortcuts :
- **UD Framework**
- **UgDuck Framework**

Additional arguments:
- It also starts with a U (like some Ubiquity classes).
- The duck has a certain capital of sympathy.

![image](https://user-images.githubusercontent.com/2511052/83359975-38a89300-a37e-11ea-9aa2-174b112bf22d.png)


[/edit]

**Comment by jcheron** [comment]
@zilveer I know that **Ugly-Duckling** is not a good idea, it's relatively unpronounceable (for the French anyway)...
But this duck is nice

**Comment by zilveer** [comment]
@jcheron then lets call it "Duck Framework" :)

**Comment by zilveer** [comment]
Also the Umbrella with the Duck was a real good logo ;)

**Comment by n0nag0n** [comment]
I approve of Duck Framework, not that my approval is needed :) Maybe it could stand for something?

**D**evelop, Design
**U**ltra, Under
**C**ool, Crazy, Cozy, 
**K**ode, Krap, Kickstarts, Knowledge, Keyboard
etc

I also would suggest the logo have colors other than.......dark stormclouds of doom and gloom greys and blacks.

**Comment by jcheron** [comment]
Thank you for your feedback @n0nag0n @zilveer

There is no framework with the name **Duck**, but there are quite a few projects with that name on gitHub ([repos with Duck](https://github.com/search?q=duck&type=Repositories)).
There's even a [Duckphp](https://github.com/dvaknheo/duckphp).

What if we leave the **U** in front?
- Reference to the old name (Ubiquity) -> so as not to disturb the users too much
- **UDuck** may also refer to the ugly duckling.

With a little less dark in the duck
![image](https://user-images.githubusercontent.com/2511052/83470684-19475e00-a483-11ea-8aaf-2abfaebcb4d1.png)

**Comment by zilveer** [comment]
@jcheron UDuck Framework sounds really good.
Regarding the logo I am the one who prefers a simple logo, the duck image can be as you have posted.
The U letter could be a little more brighter, not so dark in other words. 
Or else I like it! ;)

**Comment by n0nag0n** [comment]
Awesome! Love it!

**Comment by jcheron** [comment]
Great, we got a good one!
We'll take it in stages then, starting with slowly introducing a duck 🦆  

I'm not exactly sure how to do that renaming.
It's not enough to rename the github repositories, there's also the site address, the SEO, the documentation address, the composer packages... and I certainly have to forget some.

I think we should hire a community manager.

For the graphic design, I don't know if @richardbmx is always willing and available, nor if he is knowledgeable about ducks?

**Comment by UlasSAYGINIM** [comment]
why it is like negative thinks black and ugly. why it does not show something positive and more cheerful name? it is like lame duck :/

**Comment by ashokkumar88** [comment]
+1 for the name **Uby** framework

**Comment by alnux** [comment]
I was reading this post and if the community is not closed to just ducks, I was thinking of a very agile little feline that is the only one that can rotate its ankles 180 degrees. so I thought why not name the framework as "wiedii Framework". [here](https://en.wikipedia.org/wiki/Margay#:~:text=The%20margay%20(Leopardus%20wiedii)%20is,primary%20evergreen%20and%20deciduous%20forest.) the detail of this interesting feline.

regards
